test/articles/update.test.js

```diff
@@ -0,0 +1,179 @@
+  userToken2 = response2.body.token;
+});
+
+describe('Create Article Test', () => {
+  context(
+    `when a registered user  enter article with
+    valid details with status as draft`,
+    () => {
+      it('create and draft the article successfully', async () => {
+        const response = await chai
+          .request(app)
+          .post(`${BASE_URL}/articles/create`)
+          .set('Authorization', userToken)
+          .send(ArticleData4);
+        slug1 = response.body.slug;
+        expect(response.status).to.equal(200);
+        expect(response.body.publishedAt).to.equal(null);
+      });
+    }
+  );
+  context(
+    `when a registered user  enter article with
+    valid details with status as publish`,
+    () => {
+      it('create and publish the article successfully', async () => {
+        const response = await chai
+          .request(app)
+          .post(`${BASE_URL}/articles/create`)
+          .set('Authorization', userToken)
+          .send(updateArticleData);
+        slug2 = response.body.slug;
+        expect(response.status).to.equal(200);
+      });
+      it('mocks image upload sucessfully', async () => {
+        const uploaderstub = sinon
+          .stub(uploader, 'upload')
+          .callsFake(() => data.uploaded);
+        await ArticleController.updateArticle(request, res);
+        expect(res.statusCode).to.equal(404);
+        uploaderstub.restore();
+      });
+    }
+  );
+});
+describe('Update Article Test', () => {
+  context('when a registered enter valid details', () => {
+    it('update his article successfully', async () => {
+      const response = await chai
+        .request(app)
+        .patch(`${BASE_URL}/articles/update?slug=${slug1}`)
+        .set('Authorization', userToken)
+        .send(ArticleData2);
+      expect(response.status).to.equal(200);
+      expect(response.body.publishedAt).to.not.equal(null);
+    });
+  });
+  context(
+    'when a registered user update an article that does not exist',
+    () => {
+      it('returns not found', async () => {
+        const response = await chai
+          .request(app)
+          .patch(`${BASE_URL}/articles/update?slug=${'unknown'}`)
+          .set('Authorization', userToken)
+          .send(ArticleData2);
+        expect(response.status).to.equal(404);
+        expect(response.body.message).to.equal('Article not found');
+      });
+    }
+  );
+  context('when a registered user enter valid details', () => {
+    it('update his article successfully', async () => {
+      const response = await chai
+        .request(app)
+        .patch(`${BASE_URL}/articles/update?slug=${slug2}`)
+        .set('Authorization', userToken)
+        .send(ArticleData4);
+
+      expect(response.status).to.equal(200);
+      expect(response.body.publishedAt).to.equal(null);
+    });
+  });
+  context(
+    'when a registered enter valid details for update of article he didnt create',
```

Comment:
I believe you missed 'user' in the context of this test.

Reply:
> I believe you missed 'user' in the context of this test.

yes i did fixed thanks